client/blocks/login/login-form.jsx

```diff
@@ -217,6 +215,9 @@ export class LoginForm extends Component {
 		} );
 		this.onChangeField( event );
 		this.debouncedEmailSuggestion( event.target.value );
+		this.resetLastUsedAuthenticationMethod();
+		this.props.cancelSocialAccountConnectLinking();
+		this.props.resetAuthAccountType();
```

Comment:
I see that we're using this to remove the error message when the user is typing their username again.

It's probably not an issue, but I'm wondering if ideally we'd have both setting the state of emailSuggestion errors + these three new lines into a `onFocus` event [for the <FormTextInput](https://github.com/Automattic/wp-calypso/blob/fix/social-login-error-handling/client/blocks/login/login-form.jsx#L739) component.

Reply:
Paulo, can you elaborate a bit? Do you mean that we should do the same form state reset `onFocus` as well? Can you give me an example of how you envision the form to work after the change?

Reply:
This code runs every time the user types in the email input, which seems unnecessary.

If the intention is simply to remove the error message when the user enters a new email, we could instead clear it only when the input gains focus (using the onFocus event).